src/pages/Aggregator/Aggregator.hooks.ts

```diff
@@ -95,6 +100,35 @@ export const useAggregatorDropdowns = (
   };
 };
 
+export const useConnectedChain = (
+  startingChain: ChainEnum | '',
+  setValue: UseFormSetValue<AggregatorFormValues>
+) => {
+  const connectedChain = useSelector(chainIdSelector);
+  const provider = useSelector(providerSelector);
+
+  const wrongChainConnectedError = startingChain !== connectedChain;
+
+  useEffect(() => {
+    if (startingChain && wrongChainConnectedError) {
+      provider?.send('wallet_switchEthereumChain', [
+        { chainId: `0x${startingChain.toString(16)}` },
```

Comment:
I think you can use hexlify here:

import { utils } from 'ethers';
chainId: utils.hexlify(startingChain)

Reply:
Tried it, but theres a problem with `ethereum mainnet`
![Zrzut ekranu z 2022-03-22 14-12-29](https://user-images.githubusercontent.com/38423054/159489765-ce7db5b1-fc1f-4527-9d39-a28ddc586feb.png)

Reply:
> Nice! Remember to add some tests for the `useConnectedChain` when we will have more free time in the project

Added as todo on Clickup